src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -203,15 +213,7 @@ private SwiftCompileStep makeCompileStep(SourcePathResolverAdapter resolver) {
       compilerCommand.add("-import-underlying-module");
     }
 
-    Function<FrameworkPath, Path> frameworkPathToSearchPath =
-        CxxDescriptionEnhancer.frameworkPathToSearchPath(cxxPlatform, resolver);
-
-    compilerCommand.addAll(
-        Streams.concat(frameworks.stream(), cxxDeps.getFrameworkPaths().stream())
-            .filter(x -> !x.isSDKROOTFrameworkPath())
-            .map(frameworkPathToSearchPath)
-            .flatMap(searchPath -> ImmutableSet.of("-F", searchPath.toString()).stream())
-            .iterator());
+    frameworksArg.appendToCommandLine(compilerCommand::add, resolver);
```

Comment:
@milend , I'm not sure if I understand you correctly regarding using a `frameworksArg` to replace both `cxxPlatform` and `frameworks`.

This usage of `frameworksArg.appendToCommandLine` isn't quite consistent with the other `compilerCommand.addAll` calls nearby. I would prefer to still use the same old `compilerCommand.addAll` here for this `frameworksArg` as well.

My proposal is:
Instead of passing in one single `frameworksArg`, we pass in `Function<FrameworkPath, Path> frameworkPathToSearchPath` and `ImmutableSortedSet<FrameworkPath> frameworkPaths` into `SwiftCompile`, and mark both as `@AddToRuleKey`.

Reply:
@xianwen:

> Instead of passing in one single frameworksArg, we pass in Function<FrameworkPath, Path> frameworkPathToSearchPath and ImmutableSortedSet<FrameworkPath> frameworkPaths into SwiftCompile, and mark both as @AddToRuleKey.

The type of `frameworkPathToSearchFunction` should be `AddsToRuleKeyFunction<FrameworkPath, Path>`, agreed about the rest.

We basically have to go down that route because of the usage of `frameworkPathToSearchPath` in the `toToolFlags()` call.